File: android/app/src/main/java/com/liuhx/jswebview/injector/DefaultHandler.java
package com.liuhx.jswebview.injector;

import android.webkit.WebView;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author liuhx
 * @date 2018/7/12.
 */

public class DefaultHandler implements JsHandler {
    @Override
    public String action() {
        return null;
    }

    @Override
    public void handleJs(WebView webView, JSONObject object) {
        JSONObject toJs = new JSONObject();
        try {
            toJs.put("result", "unsupported");
        } catch (JSONException e) {
            e.printStackTrace();
            return;
        }
        WebViewInjector.invokeCallback(webView, object, toJs);
    }
}

File: ios/JsWebView/JsFramework/DefaultHandler.m
//
//  DefaultHandler.m
//  JsWebView
//
//  Created by liuhx on 2018/7/12.
//  Copyright © 2018 hursing. All rights reserved.
//

#import "DefaultHandler.h"

@implementation DefaultHandler

+ (instancetype)sharedInstance {
    static DefaultHandler *sharedInstance = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        sharedInstance = [[self alloc] init];
    });
    return sharedInstance;
}

- (NSString *)action {
    return nil;
}

- (void)handleJsFromWebView:(WKWebView *)webView info:(NSDictionary *)info {
    NSMutableDictionary *toJs = [[NSMutableDictionary alloc] initWithCapacity:3];
    toJs[@"result"] = @"unsupported";
    invokeCallback(webView, info, toJs);
}

@end
